frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -176,7 +176,9 @@ export function InstantSearchProvider({
             const deviceHandle = pathParts.length >= 2 ? pathParts[1] : '';
             const itemType = pathParts.length >= 3 ? pathParts[2] : '';
 
-            const { q, p, filter } = qsModule.parse(location.search.slice(1));
+            const { q, p, filter } = qsModule.parse(location.search, {
+               ignoreQueryPrefix: true,
+            });
```

Comment:
I wonder if it would make sense to do something like `const {q, p, filter, ...otherParams} = ` here and then reference otherParams in createURL. I worry that defaulting the query in createURL is going to prevent removing params that should be removed, since they only get overridden when they're set I think.

Reply:
Good catch. We need to unset keys of `RouteState` (ie `q`, `p`, and `filter`) in the `initialQuery`, otherwise they'll never vanish. I'm trying to think of a way to not make `createURL` depend on `otherParams` from `parseURL`, but maybe that's fine and the simplest way to do it.